scripts/export_environment.py

```diff
@@ -170,32 +160,45 @@ def resolve_env(env: Environment) -> Environment:
     >>> resolve_env({'x': '{y}', 'y': '42'})
     {'x': '42', 'y': '42'}
 
-    Literal curly braces need to be escaped:
+    Transitive references are supported:
 
-    >>> resolve_env({'x': '{{', 'y': '}}', 'z': '{{x}}'})
-    {'x': '{', 'y': '}', 'z': '{x}'}
+    >>> resolve_env({'x': '{y}', 'y': '{z}', 'z': '42'})
+    {'x': '42', 'y': '42', 'z': '42'}
 
-    Does not support transitive references:
+    Generative references formed during resolution are allowed provided that
+    curly braces are escaped:
 
-    >>> resolve_env({'x': '{y}', 'y': '{z}', 'z': '42'})
-    {'x': '{z}', 'y': '42', 'z': '42'}
+    >>> resolve_env({'x': '{{{y}}}', 'y': 'z', 'z': '42'})
+    {'x': '42', 'y': 'z', 'z': '42'}
 
-    Transitive resolution would yield
+    Compared to:
 
-    {'x': '42', 'y': '42', 'z': '42'}
+    >>> resolve_env({'x': '{{y}}', 'y': 'z', 'z': '42'})
+    {'x': 'z', 'y': 'z', 'z': '42'}
 
-    Does not support generative ones (references formed during resolution):
+    References to a missing key yields an empty string replacement:
 
-    >>> resolve_env({'x': '{y}}}', 'y': '{{z', 'z': '42'})
-    {'x': '{{z}', 'y': '{z', 'z': '42'}
+    >>> resolve_env({'x': 'y is {y}'})
+    {'x': 'y is '}
 
-    which could be
+    Unmatched curly braces are not allowed:
```

Comment:
So how _do_ I embed a single, literal, unmatched curly brace in a value?

Reply:
In my original solution this was not possible. In my current implementation any unmatched curly braces or curly braces that do not wrap a valid syntax key in the mapping will be ignored.